File: connection-backend/src/test/java/de/florianschmitt/rest/base/ChangeReferencialIntegrityOperation.java
package de.florianschmitt.rest.base;

import java.sql.SQLException;

import org.dbunit.DatabaseUnitException;
import org.dbunit.database.DatabaseConfig;
import org.dbunit.database.IDatabaseConnection;
import org.dbunit.database.statement.IBatchStatement;
import org.dbunit.database.statement.IStatementFactory;
import org.dbunit.dataset.IDataSet;
import org.dbunit.operation.DatabaseOperation;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
class ChangeReferencialIntegrityOperation extends DatabaseOperation {

    private final boolean enable;

    @Override
    public void execute(IDatabaseConnection connection, IDataSet dataSet) throws DatabaseUnitException, SQLException {
        DatabaseConfig databaseConfig = connection.getConfig();
        IStatementFactory statementFactory = (IStatementFactory) databaseConfig
                .getProperty(DatabaseConfig.PROPERTY_STATEMENT_FACTORY);
        IBatchStatement statement = statementFactory.createBatchStatement(connection);
        try {
            statement.addBatch(String.format("SET REFERENTIAL_INTEGRITY %s", enable ? "TRUE" : "FALSE"));
            statement.executeBatch();
            statement.clearBatch();
        } finally {
            statement.close();
        }
    }

    public static ChangeReferencialIntegrityOperation enable() {
        return new ChangeReferencialIntegrityOperation(true);
    }

    public static ChangeReferencialIntegrityOperation disable() {
        return new ChangeReferencialIntegrityOperation(false);
    }
}

File: src/test/kotlin/de/florianschmitt/base/ChangeReferencialIntegrityOperation.kt
package de.florianschmitt.base

import org.dbunit.DatabaseUnitException
import org.dbunit.database.DatabaseConfig
import org.dbunit.database.IDatabaseConnection
import org.dbunit.database.statement.IStatementFactory
import org.dbunit.dataset.IDataSet
import org.dbunit.operation.DatabaseOperation

import java.sql.SQLException

internal class ChangeReferencialIntegrityOperation private constructor(private val enable: Boolean) : DatabaseOperation() {

    @Throws(DatabaseUnitException::class, SQLException::class)
    override fun execute(connection: IDatabaseConnection, dataSet: IDataSet) {
        val databaseConfig = connection.config
        val statementFactory = databaseConfig
                .getProperty(DatabaseConfig.PROPERTY_STATEMENT_FACTORY) as IStatementFactory
        val statement = statementFactory.createBatchStatement(connection)
        try {
            statement.addBatch(String.format("SET REFERENTIAL_INTEGRITY %s", if (enable) "TRUE" else "FALSE"))
            statement.executeBatch()
            statement.clearBatch()
        } finally {
            statement.close()
        }
    }

    companion object {

        fun enable(): ChangeReferencialIntegrityOperation {
            return ChangeReferencialIntegrityOperation(true)
        }

        fun disable(): ChangeReferencialIntegrityOperation {
            return ChangeReferencialIntegrityOperation(false)
        }
    }
}